src/webgpu/api/validation/render_pipeline/inter_stage.spec.ts

```diff
@@ -1,10 +1,337 @@
+  .params(u => u.combine('isAsync', [false, true]))
+  .fn(async t => {
+    const { isAsync } = t.params;
+
+    const descriptor = t.getDescriptorWithStates(
+      t.getVertexStateWithOutputs(['@location(0) vout0: f32']),
+      t.getFragmentStateWithInputs(['@location(1) fin1: f32'])
+    );
+
+    t.doCreateRenderPipelineTest(isAsync, false, descriptor);
+  });
+
+g.test('location,superset')
+  .desc(`Tests that validation should fail when vertex output is a superset of fragment input.`)
+  .params(u => u.combine('isAsync', [false, true]))
+  .fn(async t => {
+    const { isAsync } = t.params;
+
+    const descriptor = t.getDescriptorWithStates(
+      t.getVertexStateWithOutputs(['@location(0) vout0: f32', '@location(1) vout1: f32']),
+      t.getFragmentStateWithInputs(['@location(1) fin1: f32'])
+    );
+
+    t.doCreateRenderPipelineTest(isAsync, false, descriptor);
+  });
+
+g.test('location,subset')
+  .desc(`Tests that validation should fail when vertex output is a subset of fragment input.`)
+  .params(u => u.combine('isAsync', [false, true]))
+  .fn(async t => {
+    const { isAsync } = t.params;
+
+    const descriptor = t.getDescriptorWithStates(
+      t.getVertexStateWithOutputs(['@location(0) vout0: f32']),
+      t.getFragmentStateWithInputs(['@location(0) fin0: f32', '@location(1) fin1: f32'])
+    );
+
+    t.doCreateRenderPipelineTest(isAsync, false, descriptor);
+  });
+
+g.test('type')
+  .desc(
+    `Tests that validation should fail when type of vertex output and fragment input at the same location doesn't match.`
+  )
+  .params(u =>
+    u.combine('isAsync', [false, true]).combineWithParams([
+      { output: 'f32', input: 'f32' },
+      { output: 'i32', input: 'f32' },
+      { output: 'u32', input: 'f32' },
+      { output: 'u32', input: 'i32' },
+      { output: 'i32', input: 'u32' },
+      { output: 'vec2<f32>', input: 'vec2<f32>' },
+      { output: 'vec3<f32>', input: 'vec2<f32>' },
+      { output: 'vec2<f32>', input: 'vec3<f32>' },
+      { output: 'vec2<f32>', input: 'f32' },
+      { output: 'f32', input: 'vec2<f32>' },
+    ])
+  )
+  .fn(async t => {
+    const { isAsync, output, input } = t.params;
+
+    const descriptor = t.getDescriptorWithStates(
+      t.getVertexStateWithOutputs([`@location(0) @interpolate(flat) vout0: ${output}`]),
+      t.getFragmentStateWithInputs([`@location(0) @interpolate(flat) fin0: ${input}`])
+    );
+
+    t.doCreateRenderPipelineTest(isAsync, output === input, descriptor);
+  });
+
+g.test('interpolation_type')
+  .desc(
+    `Tests that validation should fail when interpolation type of vertex output and fragment input at the same location doesn't match.`
+  )
+  .params(u =>
+    u.combine('isAsync', [false, true]).combineWithParams([
+      // default is @interpolate(perspective, center)
+      { output: '', input: '' },
+      { output: '', input: '@interpolate(perspective) ', _success: true },
```

Comment:
uNit: I'd trim off the trailing whitespace from these, just to make this a little more straightforward. (For example it would be confusing if you had one that was missing the space, the test would expect failure but the validation would pass.) Any extra whitespace in the generated output isn't important, of course.